src/lib.rs

```diff
@@ -195,3 +195,14 @@ impl<'a, N: 'a> Iterator for NodesMut<'a, N> {
         self.range.next()
     }
 }
+
+/// Iterator over edge _references_, which keep track of the source and
+/// target.
+#[derive(Debug)]
+pub struct EdgeReference<'a, NodeId, EdgeId> {
+    pub source: &'a NodeId,
+    pub target: &'a NodeId,
+    pub id: &'a EdgeId,
+}
+
+pub type EdgeReferences<'a, N, E> = Vec<EdgeReference<'a, N, E>>;
```

Comment:
I think this abstraction is fine, maybe a shorter name would be nice, eg `EdgeRefs` or just `Edges`?